crates/consensus/debug-client/src/client.rs

```diff
@@ -92,27 +68,67 @@ impl RpcConsensusClient {
             )
             .await
             .unwrap();
+
+            previous_block_hashes.push(block_hash);
+
+            // Load previous block hashes. We're using (head - 32) and (head - 64) as the safe and
+            // finalized block hashes.
+            // todo: chdeck for off by ones in offset
+            let safe_block_hash = get_or_fetch_previous_block(
```

Comment:
this should be a function of self

Reply:
updated, if understood this comment correctly. if not — ping again pls.